pkg/compactor/block_upload.go

```diff
@@ -394,6 +412,98 @@ func (c *MultitenantCompactor) uploadMeta(ctx context.Context, logger log.Logger
 	return nil
 }
 
+func (c *MultitenantCompactor) validateBlock(ctx context.Context, w http.ResponseWriter, blockID ulid.ULID,
+	userBkt objstore.Bucket, meta metadata.Meta) error {
+	blockDir, err := os.MkdirTemp(filepath.Join(c.compactorCfg.DataDir), "upload")
+	if err != nil {
+		return errors.Wrap(err, "failed to create temp dir")
+	}
+	defer func() {
+		if err := os.RemoveAll(blockDir); err != nil {
+			level.Warn(c.logger).Log("msg", "failed to remove temp dir", "path", blockDir, "err", err)
+		}
+	}()
+
+	rps := newRequestProgressSender(w)
```

Comment:
We should reuse "requestProgressSender" created by `completeBlockUpload`. Each time we call `newRequestProgressSender`, it currently sets Status code and Content-type. That's not necessary.